presto-native-execution/presto_cpp/main/SessionProperties.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "presto_cpp/main/SessionProperties.h"
+#include "velox/core/QueryConfig.h"
+
+using namespace facebook::presto;
+using namespace facebook::velox;
+
+namespace facebook::presto {
+
+namespace {
+const std::string boolToString(bool value) {
+  return value ? "true" : "false";
+}
+} // namespace
+
+json SessionProperty::serialize() {
+  json j;
+  j["name"] = name_;
+  j["description"] = description_;
+  j["typeSignature"] = type_;
+  j["defaultValue"] = defaultValue_;
+  j["hidden"] = hidden_;
+  return j;
+}
+
+void SessionProperties::addSessionProperty(
+    const std::string& name,
+    const std::string& description,
+    const TypePtr& type,
+    bool isHidden,
+    const std::string& veloxConfigName,
+    const std::string& veloxDefault) {
+  sessionProperties_[name] = std::make_shared<SessionProperty>(
+      name,
+      description,
+      type->toString(),
+      isHidden,
+      veloxConfigName,
+      veloxDefault);
+}
+
+// List of native session properties is kept as the source of truth here.
+SessionProperties::SessionProperties() {
+  // Use empty instance to get default property values.
+  velox::core::QueryConfig c{{}};
+  using velox::core::QueryConfig;
```

Comment:
Nit : Move this line above line 57, so that c{()} needn't use the fully qualified name.